ui/src/hooks/useTerminalConnection.ts

```diff
@@ -423,6 +423,10 @@ export const useTerminalConnection = (lunaId: Ref<string>, origin: Ref<string>)
 
       socket.send(formatMessage(terminalId.value, FORMATTER_MESSAGE_TYPE.TERMINAL_DATA, processedData));
     });
+
+    terminal.attachCustomKeyEventHandler((e: KeyboardEvent) => {
+      return handleCustomKey(e, terminal, lunaId.value, origin.value);
+    });
   };
 
   return {
```

Comment:
It is difficult to determine if there are any irregularities or potential issues in this code without more context. However, I would suggest improving the error handling of `sendEventToLuna` function. It seems like an asynchronous operation but it doesn't handle errors properly.

As for optimization suggestions:

1. Remove unnecessary imports at top.
2. Use async syntax instead.
3. Optimize file path using webpack with asset loader.
4. Consider removing unminified files after production builds.